### PR TITLE
chore: adds documentation for fledge

### PR DESCRIPTION
**Pull request by tobiasdossinger**
- adds docs for cli and getting-started
- restructures folder structure for pages url
- adds docs for config section

### CODE REVIEW

File: docs/src/guide/cli.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+```bash
+fledge create my-new-project # or fledge my-new-project -c -s ./my-stack-of-choice
+```
+
+::: tip
+For detailed guidance on how to configure your own stack, please refer to the [configuration](/config) section.
+:::
+
+## Fledge Set
```
Comment (schapka) [inline docs/src/guide/cli.md]: This should be Config.

The CLI provides a sub program (config) which has two commands: `set` and `get`

File: docs/package.json [hunk 2]
```diff
@@ -10,12 +10,19 @@
...
+    "zod": "^3.23.8",
+    "zod-to-ts": "^1.2.0"
   },
   "scripts": {
     "docs:dev": "vitepress dev",
     "docs:build": "vitepress build",
-    "docs:preview": "vitepress preview"
+    "docs:preview": "vitepress preview",
+    "docs:lint": "eslint . --fix && prettier -w .",
+    "docs:generate-types": "node --experimental-strip-types ./scripts/zod-to-types.ts"
```
Comment (schapka) [inline docs/package.json]: Maybe we could try to use Build-Time Data Loading for loading this data:
https://vitepress.dev/guide/data-loading#build-time-data-loading

Comment (tobiasdossinger) [reply docs/package.json]: just changed it: https://github.com/schapka/fledge/pull/1/commits/43bbe38240b70377e71a90a9dc5de07c1200de0d

File: docs/data/zod.data.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { defineLoader } from 'vitepress';
+import { printNode, zodToTs } from 'zod-to-ts';
+
+import { configSchema, type Config } from '../../src/core/schemas.js';
+
+type Data = {
+  configSchema: string;
+};
+
+function convertZodSchemaToString(schema: ZodTypeAny | Config, schemaTitle: string) {
```
Comment (schapka) [inline docs/data/zod.data.ts]: I assume it should work to just use `typeof configSchema` and we should also be able to get rid of configSchema as unknown as Config afterwards.

Comment (tobiasdossinger) [reply docs/data/zod.data.ts]: due to zodToTs is first parameter type ZodTypeAny, like this? 
```suggestion
function convertZodSchemaToString(schema: ZodTypeAny | typeof configSchema, schemaTitle: string) { 
const { node } = zodToTs(schema as ZodTypeAny, schemaTitle);
  return printNode(node);
}
```

**Pull request merged**